few-shot示例保留向量查询最佳匹配结果

chsumu opened the pull request:
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Additional information

Any additional information, configuration or data that might be necessary to reproduce the issue.

chsumu committed:
(improvement)(launcher) The few-shot example is best reserved for vector queries

chsumu committed:
Merge branch 'tencentmusic:master' into master

chsumu committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	common/src/main/java/com/tencent/supersonic/common/service/impl/ExemplarServiceImpl.java
#	headless/chat/src/main/java/com/tencent/supersonic/headless/chat/parser/llm/PromptHelper.java

chsumu committed:
(improvement)(launcher) The few-shot example is best reserved for vector queries(#1417)

jerryjzhang commented on chat/server/src/main/java/com/tencent/supersonic/chat/server/service/impl/MemoryServiceImpl.java:
```diff
@@ -91,8 +92,10 @@ public List<ChatMemoryDO> getMemories(ChatMemoryFilter chatMemoryFilter) {
+                .eq(ChatMemoryDO::getStatus, MemoryStatus.PENDING)
+                .eq(ChatMemoryDO::getLlmReviewRet, MemoryReviewResult.POSITIVE)
+                .eq(ChatMemoryDO::getHumanReviewRet, MemoryReviewResult.POSITIVE);
         return chatMemoryRepository.getMemories(queryWrapper);
```
这里需要获取所有pending的memory交给LLM做review，如果加这个状态过滤逻辑就不对了

chsumu replied:
这里按原有的逻辑实测会污染向量库，LLM 会判断错误，还是建议增加管理员判断逻辑

chsumu replied:
抱歉，我的错误，应该在 MemoryReviewTask 里面做过滤逻辑的

jerryjzhang reviewed (commented):
增加score是好的想法，只不过当前如果无法从memory里召回到few shot样例，就会采用系统默认的样例，相比较起来虽然memory样例有可能语义不太相关，但总归是属于同一个domain，会比系统默认样例要更合适一些

chsumu committed:
Update MemoryServiceImpl.java

llm review

chsumu committed:
fix(vector): 避免向量数据库重复插入数据

增加指标和维度 is_vector 字段，用于标识数据是否已向量
增加指标向量化逻辑判断，只有发布(is_publish)后的指标才会向量
增加是否初始化 s2-exemplar.json(text2dsl_agent_collection) 判断逻辑
增加向量化定时任务（条件：开启管理员评估, 未开启走原有大模型评估逻辑）

chsumu committed:
fix(vector): Prevents repeated insertion of data into the vector database

Add the Metric and Dimension is_vector fields to identify whether the data is vectored or not
Added indicator vectorization logic judgment, and only the indicator after the release (is_publish) will be vectorized
Add the s2-exemplar.json (text2dsl_agent_collection) judgment logic for initialization
Add a vectorized scheduled task (condition: enable administrator evaluation, but not enable the original large model evaluation logic)

chsumu committed:
fix(vector): Prevents repeated insertion of data into the vector database

Add the Metric and Dimension is_vector fields to identify whether the data is vectored or not
Added indicator vectorization logic judgment, and only the indicator after the release (is_publish) will be vectorized
Add the s2-exemplar.json (text2dsl_agent_collection) judgment logic for initialization
Add a vectorized scheduled task (condition: enable administrator evaluation, but not enable the original large model evaluation logic)

chsumu committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	chat/server/src/main/java/com/tencent/supersonic/chat/server/memory/MemoryVectorTask.java
